processing/src/main/java/org/apache/druid/query/groupby/GroupByQuery.java

```diff
@@ -207,9 +210,17 @@ private GroupByQuery(
       Preconditions.checkArgument(spec != null, "dimensions has null DimensionSpec");
     }
 
+    List<String> dimensionNames = this.dimensions
+        .stream()
+        .map(DimensionSpec::getOutputName)
+        .collect(Collectors.toList());
+    String timestampField = getContextValue(CTX_TIMESTAMP_RESULT_FIELD);
+    if (timestampField != null) {
+      dimensionNames.add(timestampField);
```

Comment:
I don't think groupBy query engine should know about the timestamp column remapping at all. It should be able to use the `__time` column as it is and the sql layer should handle the column remapping.

Reply:
the timestampField maybe referenced by the postAgg, so the timestampField should be added back to the dependencies of the postAggs here, otherwise the constructor will fail